doc/source/whatsnew/v0.24.0.rst

```diff
@@ -798,6 +798,34 @@ Bug in :func:`DataFrame.to_dict` raises ``ValueError`` when used with
 
     df.to_dict(orient='index')
 
+.. _whatsnew_0240.api_breaking.incompatible_index_unions
+
+Incompatible Index Type Unions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When performing an :func:`Index.union` operation, between objects of incompatible dtypes,
+the result will be a base :class:`Index` of dtype `object`. This behavior holds true for
+unions between :class:`Index` objects that previously would have been prohibited. The dtype
+of empty :class:`Index` objects will now be evaluated before performing union operations
+rather than simply returning the other :class:`Index` object (:issue:`23525`).
+
+
+Previous Behavior:
+
+.. ipython:: python
+
+    In [1]: pd.period_range('19910905', periods=2).union(pd.Int64Index([1, 2, 3]))
+
+    In [2]: pd.Index([]).astype(str).union(pd.Index([1, 2, 3]))
+
+Current Behavior:
+
+.. ipython:: python
+
+    In [1]: pd.period_range('19910905', periods=2).union(pd.Int64Index([1, 2, 3]))
```

Comment:
this is ok, but don't neeed the ipython prompts

also use ``*`` around the ``Current`` and ``Previous``, see other sub-sections for how

Reply:
fixed.